packages/engine/src/lib/operations.ts

```diff
@@ -31,28 +31,20 @@ import { testExecutionContext } from './handler/context/test-execution-context';
 import { flowExecutor } from './handler/flow-executor';
 import { blockHelper } from './helper/block-helper';
 import { triggerHelper } from './helper/trigger-helper';
-import { progressService } from './services/progress.service';
 import { utils } from './utils';
 
 const executeFlow = async (
   input: ExecuteFlowOperation,
   context: FlowExecutorContext,
 ): Promise<EngineResponse<Pick<FlowRunResponse, 'status' | 'error'>>> => {
   const constants = EngineConstants.fromExecuteFlowInput(input);
-  const output = await flowExecutor.execute({
-    action: input.flowVersion.trigger.nextAction,
+
+  const response = await flowExecutor.triggerFlowExecutor({
```

Comment:
1. I think we should keep it as "execute", as before
2. move 
```
await progressService.sendUpdate({
    engineConstants: constants,
    flowExecutorContext: newContext,
  });
```
to be there always
3. this should then just be
```
  const newContext =
    output.verdict === ExecutionVerdict.RUNNING
      ? output.setVerdict(ExecutionVerdict.SUCCEEDED, output.verdictResponse)
      : output;
  const response = await newContext.toResponse();
```

wdyt?